lib/screen/connection/persona_connections_page.dart

```diff
@@ -111,27 +123,108 @@ class _PersonaConnectionsPageState extends State<PersonaConnectionsPage>
     }
   }
 
+  void _showConnectionOption() {
+    final options = [
+      OptionItem(
+          title: "connect_via_clipboard".tr(),
+          icon: SvgPicture.asset("assets/images/DApp.svg"),
+          onTap: () async {
+            if (!mounted) return;
+            Navigator.of(context).pop();
+            try {
+              final clipboardData = await Clipboard.getData("text/plain");
+              if (clipboardData == null ||
+                  clipboardData.text == null ||
+                  clipboardData.text!.isEmpty) {
+                throw ConnectionViaClipboardError("Clipboard is empty");
+              }
+              final text = clipboardData.text!;
+              log.info("Connect via clipboard: $text");
+              await _processDeeplink(text);
+            } catch (e) {
+              log.info("Connect via clipboard: failed ${e.toString()}");
+              if (e is ConnectionViaClipboardError) {
+                if (!mounted) return;
+                UIHelper.hideInfoDialog(context);
+                UIHelper.showInvalidURI(context);
+              }
+            }
+          }),
+      OptionItem(),
+    ];
+    UIHelper.showDrawerAction(context, options: options);
+  }
+
+  bool _isTezosBeconUri(String uri) {
+    try {
+      final base58Decode = bs58check.decode(uri);
+      final uriData = jsonDecode(String.fromCharCodes(base58Decode));
+      return uriData['type'] == "p2p-pairing-request";
+    } catch (_) {
+      return false;
+    }
+  }
+
+  bool _isWalletConnectUri(String uri) {
+    return uri.startsWith("wc:");
+  }
+
+  bool _isUriValid(String uri) {
+    return (_isWalletConnectUri(uri) || _isTezosBeconUri(uri));
+  }
+
+  Future<void> _processDeeplink(String code) async {
+    if (!_isUriValid(code)) {
+      throw ConnectionViaClipboardError("Invalid URI");
+    }
+    bool isConnected = false;
+    ValueNotifier<bool> flag;
+    if (code.startsWith("wc:")) {
+      flag = _walletConnecService.uriValid;
+      _walletConnecService.connect(code);
+    } else {
+      final tezosUri = "tezos://?type=tzip10&data=$code";
+      flag = _tezosBeaconService.uriValid;
+      await _tezosBeaconService.addPeer(tezosUri);
+      injector<NavigationService>().showContactingDialog();
+    }
+    flag.addListener(() {
```

Comment:
I think you should try to modify the code a little bit to `removeListener` when `dispose`.

Reply:
then maybe you might need to change `Future.delayed` to `Timer`, then you can also cancel it when `dispose`